libs/ledgerjs/packages/errors/src/index.ts

```diff
@@ -281,24 +281,30 @@ export function getAltStatusMessage(code: number): string | undefined | null {
  * Error thrown when a device returned a non success status.
  * the error.statusCode is one of the `StatusCodes` exported by this library.
  */
-export function TransportStatusError(statusCode: number): void {
-  const statusText =
-    Object.keys(StatusCodes).find(k => StatusCodes[k] === statusCode) || "UNKNOWN_ERROR";
-  const smsg = getAltStatusMessage(statusCode) || statusText;
-  const statusCodeStr = statusCode.toString(16);
-  const message = `Ledger device: ${smsg} (0x${statusCodeStr})`;
+export class TransportStatusError extends Error {
+  statusCode: number;
+  statusText: string;
 
-  // Maps to a LockedDeviceError
-  if (statusCode === StatusCodes.LOCKED_DEVICE) {
-    throw new LockedDeviceError(message);
-  }
+  constructor(statusCode: number) {
+    const statusText =
+      Object.keys(StatusCodes).find(k => StatusCodes[k] === statusCode) || "UNKNOWN_ERROR";
+    const smsg = getAltStatusMessage(statusCode) || statusText;
+    const statusCodeStr = statusCode.toString(16);
+    const message = `Ledger device: ${smsg} (0x${statusCodeStr})`;
+
+    super(message);
+
+    // Maps to a LockedDeviceError
+    if (statusCode === StatusCodes.LOCKED_DEVICE) {
+      throw new LockedDeviceError(message);
+    }
 
-  this.name = "TransportStatusError";
-  this.message = message;
-  this.stack = new Error(message).stack;
-  this.statusCode = statusCode;
-  this.statusText = statusText;
+    this.name = "TransportStatusError";
+    this.message = message;
+    this.stack = new Error(message).stack;
```

Comment:
do you know if this is still needed, since you are doing a super(message) already?

also, is this part also covered by tests? because i'm only aware of testing .stack on a specific error from @ledgerhq/errors but it would be great to have non reg on this TransportStatusError too (and confirm it didn't change)

https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/errors/src/index.test.ts#L20

Reply:
oh yeah it's probably not needed ! I will double check

I'll also look at the tests to see what i can add/do ;)

Reply:
Added some simple regression testing and tried to refactor the ugly `throw new LockedDeviceError` - WDYT ?